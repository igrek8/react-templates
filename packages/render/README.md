# @react-templates/render

Renders a template using React & CSS modules

## Installation

```bash
npm install @react-templates/render --save
yarn add @react-templates/render
```

## Usage

```tsx
import React from 'react';
import { IntlProvider, useIntl } from 'react-intl';
import { renderHtml } from '@react-templates/render';
import { WrapperProps } from '@react-templates/render';

// 1) Define props
interface MyTemplateProps {
  user: string;
}

// 2) Define a React template
const MyTemplate: React.FC<MyTemplateProps> = ({ user }) => {
  const intl = useIntl();
  return <p>{intl.formatMessage({ id: 'welcome' }, { user })}</p>;
};

// 3) Define localization
const localization: Record<string, Record<string, string>> = {
  en: {
    welcome: 'Hello, {user}',
  },
};

// 4) Define React providers for localization
const wrapper: React.FC<WrapperProps> = ({ locale, children }) => {
  return (
    <IntlProvider locale={locale} messages={localization[locale]}>
      {children}
    </IntlProvider>
  );
};

// 5) Render HTML
renderHtml<MyTemplateProps>({
  title: 'Hello World!',
  template: MyTemplate,
  data: { user: 'John' },
  dir: 'ltr',
  locale: 'en',
  timeZone: 'Europe/Berlin',
  baseUrl: 'https://bucket.s3.eu-central-1.amazonaws.com',
  wrapper,
});
```

```html
<!DOCTYPE html>
<html lang="en" dir="ltr">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Hello World!</title>
  </head>
  <body>
    <p>Hello, John</p>
  </body>
</html>
```
