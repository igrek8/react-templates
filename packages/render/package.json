{
  "name": "@react-templates/render",
  "version": "0.0.5",
  "main": "./dist/index.js",
  "types": "./dist/index.d.ts",
  "license": "MIT",
  "homepage": "https://github.com/igrek8/react-templates#readme",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "react-templates",
    "render"
  ],
  "description": "Renders a react template to a static markup or text",
  "bugs": {
    "url": "https://github.com/igrek8/react-templates/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/igrek8/react-templates.git#issue"
  },
  "scripts": {
    "build": "tsc",
    "dev": "tsc --watch"
  },
  "dependencies": {
    "tslib": "^2"
  },
  "peerDependencies": {
    "react": "^17 || ^18",
    "react-dom": "^17 || ^18"
  },
  "devDependencies": {
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "tslib": "2.5.2"
  }
}
