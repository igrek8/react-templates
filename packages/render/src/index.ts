export * from './Localization';
export * from './RenderHtmlOptions';
export * from './RenderTextOptions';
export * from './WrapperProps';
export * from './renderHtml';
export * from './renderText';
