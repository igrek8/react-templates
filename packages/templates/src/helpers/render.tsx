import React from 'react';
import { IntlProvider } from 'react-intl';
import { i18n } from '../i18n';

export function create(element: React.ReactElement) {
  return (
    <html>
      <head>
        <title>Snapshot</title>
      </head>
      <body>
        <IntlProvider locale={'en'} messages={i18n['en']} timeZone="UTC">
          {element}
        </IntlProvider>
      </body>
    </html>
  );
}
