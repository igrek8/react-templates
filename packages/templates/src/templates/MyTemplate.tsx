import React from 'react';
import { useIntl } from 'react-intl';
import url from '../assets/image.jpeg';
import css from './MyTemplate.module.css';

export interface MyTemplateProps {
  user: string;
}

export const MyTemplate: React.FC<MyTemplateProps> = ({ user }) => {
  const intl = useIntl();
  const now = new Date('2023-01-01');
  const timeNow = intl.formatDate(now, { dateStyle: 'long', timeStyle: 'long' });
  return (
    <>
      <p className={css.title}>{intl.formatMessage({ id: 'welcome' }, { user })}</p>
      <p>{intl.formatMessage({ id: 'now' }, { now: timeNow })}</p>
      <img src={url} alt="Image" width={200} />
    </>
  );
};
