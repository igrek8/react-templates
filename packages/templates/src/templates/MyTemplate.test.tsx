import React from 'react';
import { create } from '../helpers/render';
import { MyTemplate } from './MyTemplate';

test('renders a template', async () => {
  expect(create(<MyTemplate user="John" />)).toMatchSnapshot();
});
