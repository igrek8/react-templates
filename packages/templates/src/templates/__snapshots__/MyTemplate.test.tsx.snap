// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a template 1`] = `
<html>
  <head>
    <title>Snapshot</title>
  </head>
  <body>
    <p class="title">Hello, John!</p>
    <p>It is January 1, 2023 at 12:00:00â€¯AM UTC</p>
    <img src="image.jpeg" alt="Image" width="200" />
  </body>
</html>
`;
